pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -75,20 +75,28 @@ func (a *actuator) Reconcile(ctx context.Context, bastion *extensionsv1alpha1.Ba
 		return err
 	}
 
-	instanceType, err := aliCloudECSClient.GetInstanceType(cores)
-	if err != nil {
-		return err
-	}
+	var instanceTypeId string
+	for cores := 1; cores <= 2; cores++ {
+		instanceType, err := aliCloudECSClient.GetInstanceType(cores)
+		if err != nil {
+			return err
+		}
 
-	if instanceType == nil ||
-		len(instanceType.AvailableZones.AvailableZone) == 0 ||
-		len(instanceType.AvailableZones.AvailableZone[0].AvailableResources.AvailableResource) == 0 ||
-		len(instanceType.AvailableZones.AvailableZone[0].AvailableResources.AvailableResource[0].SupportedResources.SupportedResource) == 0 ||
-		instanceType.AvailableZones.AvailableZone[0].AvailableResources.AvailableResource[0].SupportedResources.SupportedResource[0].Status != "Available" {
-		return errors.New("one cores instance is empty or unavailable")
+		if instanceType == nil ||
+			len(instanceType.AvailableZones.AvailableZone) == 0 ||
+			len(instanceType.AvailableZones.AvailableZone[0].AvailableResources.AvailableResource) == 0 ||
+			len(instanceType.AvailableZones.AvailableZone[0].AvailableResources.AvailableResource[0].SupportedResources.SupportedResource) == 0 ||
+			instanceType.AvailableZones.AvailableZone[0].AvailableResources.AvailableResource[0].SupportedResources.SupportedResource[0].Status != "Available" {
+			continue
+		}
+
+		instanceTypeId = instanceType.AvailableZones.AvailableZone[0].AvailableResources.AvailableResource[0].SupportedResources.SupportedResource[0].Value
+		break
 	}
 
-	instanceTypeId := instanceType.AvailableZones.AvailableZone[0].AvailableResources.AvailableResource[0].SupportedResources.SupportedResource[0].Value
+	if instanceTypeId == "" {
+		instanceTypeId = cluster.CloudProfile.Spec.MachineTypes[0].Name
```

Comment:
committed in 85c3f15c,  c13fde31

> it's not guaranteed that the first entry is one with the lowest amount of CPUs

I'm trying to avoid using the default instance type when bastion creation, ideally using the lowest amount of CPUs from an available zone.  it has about ~20 different instance types from `cores <=2` when I testing depends on the available zone.  

I was thinking might not need extra logic to search the lowest amount of CPUs from cloud profile machine types list.  pick up the first one just in case.